### PR TITLE
feat: vimspector configurations picker

### PR DESCRIPTION
**Pull request by sharksforarms**
@tjdevries @rockerBOO @tami5 

Let me know if I'm missing anything!

Cheers to the first telescope plugin :tada:

### CODE REVIEW

**Comment by rockerBOO** [comment]
I think this looks good. :+1: 

In the readme can we get an install using Plug (or whichever you use). Showing the telescope.nvim dependency. Also would be nice to have a screenshot. 

Is configurations the best name?

File: lua/telescope/extensions/vimspector.lua [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    finder    = finders.new_table {
+      results = configurations
+    },
+    sorter = sorters.get_generic_fuzzy_sorter(),
+    attach_mappings = function(prompt_bufnr, map)
+      local start_debugger = function()
+        local selection = actions.get_selected_entry(prompt_bufnr)
+        actions.close(prompt_bufnr)
+
+        vim.cmd("call vimspector#LaunchWithSettings(#{ configuration: '" .. selection.value .. "'})")
```
Comment (tjdevries) [inline lua/telescope/extensions/vimspector.lua]: `vim.fn["vimspector#LaunchWithSettings"]({configuration = selection.value})` or somethin like that.

**Comment by kkharji** [comment]
So to map the function I need to access it via 
`require 'telescope.extensions.viminspector'.afunction(ots)`

Yah this a bit too long, 

`require 'telescope.viminspector'.afunction(ots)`

`require 'telescope.builtin'.afunction(ots)`

I'd say drop the extension part if @tjdevries agrees.

**Comment by sharksforarms** [comment]
> In the readme can we get an install using Plug (or whichever you use). Showing the telescope.nvim dependency. Also would be nice to have a screenshot.

Done

> Is configurations the best name?

That's the terminology used in vimspector, depends on how far we want to stray. Other options could be `launch_configurations`, `launches`, `targets`

**Comment by sharksforarms** [comment]
> So to map the function I need to access it via
> `require 'telescope.extensions.viminspector'.afunction(ots)`
> 
> Yah this a bit too long,

Perhaps `require 'telescope.ext.vimspector'.afunction(ots)` ?

**Comment by kkharji** [comment]
> > So to map the function I need to access it via
> 
> > `require 'telescope.extensions.viminspector'.afunction(ots)`
> 
> > 
> 
> > Yah this a bit too long,
> 
> 
> 
> Perhaps `require 'telescope.ext.vimspector'.afunction(ots)` ?

Basically the same as the pervious one 4 dots in total to access something. Best way to go about this is as I suggest earlier having the file under telescope directly. 

`require 'telescope.vimspector'.afunction(ots)`

`require 'telescope.builtin'.afunction(ots)`

`require 'telescope.vimwiki'.afunction(ots)`

The only reason a extension directory is needed is when there files with preserved names such as sorters, builtins etc. Even though that wouldn't be an issue, the plugin can have lib dir or a directory with same name of the plugin under telescope.

**Comment by rockerBOO** [comment]
`'telescope.vimspector'` would be my favorite. 

Of note is to keep the telescope API clean enough to not have conflicting names later. Most plugins have unique enough names.

**Comment by tjdevries** [comment]
OK, here are some of my first thoughts.

1. It is important to namespace all external plugins in an obvious way. If we do not do this, we will have to make breaking changes later when we want to do special things with "extensions".
2. It is important that when people are writing code for telescope it is obvious what is in `core` and what is not. Plugins will not be allowed (unless there is as good reason) to expose anything not first under the namespace.

What I propose instead is in `telescope.nvim` I write some wrapper code for extensions. These extensions will be imported by doing:

```
lua TeleExt = require('telescope').extensions  -- or require('telescope.extensions'), i have not decided
nnoremap <leader>vl :lua TeleExt.vimspector.launch()<CR>
```

or similar.

the `.extensions` will be a special module that allows plugins to do things like:
1. setup the extension
2. assert that certain requirements exist
3. <future ideas here>

People can write their own lua to make the mappings shorter, but it's not a good tradeoff to make mappings short and lose explicitness, namespaces and other future ideas.

(this is just my first draft of the ideas, but that's what I'm thinking).

This PR will probably need to wait til I play a bit with the idea of extensions.

**Comment by kkharji** [comment]
@tjdevries I think this is a great justification for having extra namespace.
It reminded me of how LeaderF manages and enables extensions. 

```
let s:extension = {
            \   "name": "marks",
            \   "help": "navigate the marks",
            \   "manager_id": "leaderf#Marks#managerId",
            \   "arguments": [
            \   ]
            \ }
" In order that `Leaderf marks` is available
call g:LfRegisterPythonExtension(s:extension.name, s:extension)


```
As a developer I think this is just added complexity and I would hate to be forced to write something like this, just to enable my plugin to work with telescope.nvim

**Comment by tjdevries** [comment]
Can you check latest master of telescope and let me know what you think? If it seems good, you can convert to that style of plugin/extension and then we can merge!

**Comment by sharksforarms** [comment]
@tjdevries ready for review

**Comment by tjdevries** [comment]
LGTM as a first iteration. Feel free to merge to master anything you work on that isn't controversial for the repo :)

**Pull request merged**